lib/spack/spack/filesystem_view.py

```diff
@@ -0,0 +1,494 @@
+                * ..they are not activte in the view.
+
+            Takes `with_dependencies` keyword argument so that the status of
+            dependencies is printed as well.
+        """
+        raise NotImplementedError
+
+
+def with_extensions(f):
+    """
+        Decorator for functions operating with local-extensions.
+    """
+    @ft.wraps(f)
+    def wrapped(self, *args, **kwargs):
+        with self._context_exts:
+            return f(self, *args, **kwargs)
+    return wrapped
+
+
+class YamlFilesystemView(FilesystemView):
+    """
+        Filesystem view to work with a yaml based directory layout.
+    """
+
+    def __init__(self, root, layout, **kwargs):
+        super(YamlFilesystemView, self).__init__(root, layout, **kwargs)
+
+        self.extensions_layout = YamlViewExtensionsLayout(root, layout)
+
+        from spack import store as spack_store
+        self._context_exts = ReplaceContext(parent_object=spack_store,
+                                            attribute="extensions",
+                                            replacement=self.extensions_layout)
+
+        self._croot = colorize_root(self.root) + " "
+
+    def activate(self, *specs, **kwargs):
+        assert all((s.concrete for s in specs))
+        specs = set(specs)
+
+        if kwargs.get("with_dependencies", True):
+            specs = get_dependencies(specs)
+
+        if kwargs.get("exclude", None):
+            specs = set(filter_exclude(specs, kwargs["exclude"]))
+
+        conflicts = self.get_conflicts(*specs)
+
+        if conflicts:
+            for s, v in conflicts:
+                self.print_conflict(v, s)
+            return
+
+        extensions = set(filter(lambda s: s.package.is_extension, specs))
+        standalones = specs - extensions
+
+        set(map(self._check_no_ext_conflicts, extensions))
+        # fail on first error, otherwise link extensions as well
+        if all(map(self.activate_standalone, standalones)):
+            all(map(self.activate_extension, extensions))
+
+    @with_extensions
+    def activate_extension(self, spec):
+        if not spec.package.is_extension:
+            tty.error(self._croot + 'Package %s is not an extension.'
+                      % spec.name)
+            return False
+
+        try:
+            if not spec.package.activated:
+                spec.package.do_activate(verbose=self.verbose)
+
+        except ExtensionAlreadyInstalledError:
+            # As we use sets in activate(), the order in which packages get
+            # activated is essentially random. So this spec might have already
+            # been activated as dependency of another package -> fail silently
+            pass
+
+        # make sure the meta folder is linked as well (this is not done by the
+        # extension-activation mechnism)
+        if not self.check_active(spec):
+            self.link_meta_folder(spec)
+
+        return True
+
+    def activate_standalone(self, spec):
```

Comment:
When I decided on the name I wanted to keep in the spirit of extensions in that activating a package is linking it somewhere else, but I have no strong opinion on the issue.

Your concern is very valid that it might confuse people → I'll change it.